functions/vshn-postgres-func/encrypted_pvc.go

```diff
@@ -0,0 +1,68 @@
+package vshnpostgres
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/sethvargo/go-password/password"
+
+	"github.com/go-logr/logr"
+	"github.com/vshn/appcat-comp-functions/runtime"
+	vshnv1 "github.com/vshn/component-appcat/apis/vshn/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AddPvcSecret adds a secret for the encrypted PVC for the  PostgreSQL instance.
+func AddPvcSecret(ctx context.Context, log logr.Logger, iof *runtime.Runtime, comp *vshnv1.VSHNPostgreSQL) (*vshnv1.VSHNPostgreSQL, error) {
+
+	log.Info("Check if encrypted storage is enabled")
+
+	log.V(1).Info("Transforming", "obj", iof)
+
+	// luksSecretResourceName is the resource name defined in the composition
+	// This name is different from metadata.name of the same resource
+	// The value is hardcoded in the composition for each resource and due to crossplane limitation
+	// it cannot be matched to the metadata.name
+	var luksSecretResourceName = comp.Name + "-luks-key"
+
+	encryptionSpec := comp.Spec.Parameters.Encryption
+
+	if !encryptionSpec.Enabled {
+		log.Info("Encryption not enabled")
+		return comp, nil
+	}
+
+	log.Info("Adding secret to composite")
+
+	secret := &v1.Secret{}
+
+	err := iof.GetFromObservedKubeObject(ctx, secret, luksSecretResourceName)
+	var luksKey string
+	if err != nil {
+		log.Info("Secret does not exist yet. Creating...")
+		luksKey, _ = password.Generate(64, 10, 1, false, true)
```

Comment:
Don't ignore the error.  I can see that it's unlikely to ever return an error, but if it does this could result in very weird bugs at best and a security issue at worst.

Reply:
Done

Reply:
You still ignore the error :sweat_smile:

Reply:
I don't get it. Where do I ignore the error?
I do handle the error and return a `runtime.NewFatalErr` if an error occurs.

Reply:
Well `password.Generate` returns a password and an error and you explicitly ignore the error with `_`.

This is unlikely to fail, but if it does we might set an empty or otherwise wrong password, which would be bad

Reply:
Oh I see. Fixed it by returning a `runtime.NewFatalErr` here as well if the `password.Generate` should fail.